rs/index/src/collection/collection.rs

```diff
@@ -589,6 +631,9 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
 
         self.versions.insert(new_version, toc);
 
+        // New TOC now contains the new segments. Update the metrics
+        INTERNAL_METRICS.num_active_segments_inc_by(&self.collection_name, names.len() as i64);
```

Comment:
I think you can just look at the TOC to determine number of active segments. So let's just use `set` instead of `incr_by` or `decr_by`.